File: example/java-sample/src/main/java/com/naver/namexample/sample/RewardedFragment.java
/*
 * NAM(Naver Ad Manager) SDK for Android
 *
 * Copyright 2022-present NAVER Corp.
 * All rights reserved.
 *
 * Unauthorized use, modification and redistribution of this software are strongly prohibited.
 */
package com.naver.namexample.sample;

import android.os.Bundle;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.TextView;
import androidx.fragment.app.Fragment;
import com.naver.gfpsdk.AdParam;
import com.naver.gfpsdk.GfpError;
import com.naver.gfpsdk.GfpRewardedAd;
import com.naver.gfpsdk.GfpRewardedAdManager;
import com.naver.gfpsdk.RewardedAdListener;
import com.naver.namexample.R;
import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.Locale;

public class RewardedFragment extends Fragment {
    private static final String AD_UNIT_ID = "aos_rv_test";
    private GfpRewardedAdManager rewardedAdManager;
    private Button showButton;
    private final SimpleDateFormat sdf = new SimpleDateFormat("HH:mm:ss.SSS", Locale.getDefault());
    private TextView logTextView;

    @Override
    public View onCreateView(
            LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_rewarded, container, false);
        showButton = view.findViewById(R.id.show_button);
        showButton.setEnabled(false);

        showButton.setOnClickListener(
                new View.OnClickListener() {
                    @Override
                    public void onClick(View v) {
                        // 유효성 체크
                        if (rewardedAdManager.isAdInvalidated()) {
                            logTextView.append(
                                    String.format("[%s] 광고가 유효하지 않습니다%n", sdf.format(new Date())));
                            return;
                        }

                        rewardedAdManager.showAd(requireActivity());
                    }
                });

        logTextView = view.findViewById(R.id.log_text_view);

        AdParam adParam = new AdParam.Builder().setAdUnitId(AD_UNIT_ID).build();

        rewardedAdManager = new GfpRewardedAdManager(requireActivity(), adParam);

        // 필요시 타임아웃 셋팅
        // rewardedAdManager.setTimeoutMillis(60_000L);

        rewardedAdManager.setAdListener(
                new RewardedAdListener() {
                    @Override
                    public void onAdLoaded(GfpRewardedAd ad) {
                        logTextView.append(
                                String.format(
                                        "[%s] AD Loaded.%n    AdProviderName: %s%n",
                                        sdf.format(new Date()), ad.getAdProviderName()));
                        showButton.setEnabled(true);
                    }

                    @Override
                    public void onAdStarted(GfpRewardedAd ad) {
                        logTextView.append(
                                String.format("[%s] AD Started.%n", sdf.format(new Date())));
                    }

                    @Override
                    public void onAdClicked(GfpRewardedAd ad) {
                        logTextView.append(
                                String.format("[%s] AD Clicked.%n", sdf.format(new Date())));
                    }

                    @Override
                    public void onAdClosed(GfpRewardedAd ad) {
                        logTextView.append(
                                String.format("[%s] AD Closed.%n", sdf.format(new Date())));
                        showButton.setEnabled(false);
                    }

                    @Override
                    public void onAdCompleted(GfpRewardedAd ad) {
                        logTextView.append(
                                String.format("[%s] AD Completed.%n", sdf.format(new Date())));
                    }

                    @Override
                    public void onError(GfpRewardedAd ad, GfpError error) {
                        logTextView.append(
                                String.format(
                                        Locale.US,
                                        "[%s] Error occurred.%n    code[%d]%n    subCode[%s]%n    message[%s]%n",
                                        sdf.format(new Date()),
                                        error.getErrorCode(),
                                        error.getErrorSubCode(),
                                        error.getErrorMessage()));
                        Log.e("RewardedFragment", ad.getResponseInfo().toString());
                    }
                });

        rewardedAdManager.loadAd();
        return view;
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();

        if (rewardedAdManager != null) {
            rewardedAdManager.destroy();
        }
    }
}

File: example/kotlin-sample/src/main/java/com/naver/namexample/sample/RewardedFragment.kt
/*
 * NAM(Naver Ad Manager) SDK for Android
 *
 * Copyright 2022-present NAVER Corp.
 * All rights reserved.
 *
 * Unauthorized use, modification and redistribution of this software are strongly prohibited.
 */
package com.naver.namexample.sample

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.TextView
import androidx.fragment.app.Fragment
import com.naver.gfpsdk.AdParam
import com.naver.gfpsdk.GfpError
import com.naver.gfpsdk.GfpRewardedAd
import com.naver.gfpsdk.GfpRewardedAdManager
import com.naver.gfpsdk.RewardedAdListener
import com.naver.namexample.databinding.FragmentRewardedBinding
import com.naver.namexample.util.DateUtil

class RewardedFragment : Fragment() {
    private var rewardedAdManager: GfpRewardedAdManager? = null
    private lateinit var binding: FragmentRewardedBinding
    private lateinit var logTextView: TextView
    private lateinit var showButton: Button

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        binding = FragmentRewardedBinding.inflate(inflater)
        logTextView = binding.logTextView
        showButton = binding.showButton.apply {
            isEnabled = false
            setOnClickListener(
                View.OnClickListener {
                    if (rewardedAdManager?.isAdInvalidated != false) {
                        logTextView.append("[${DateUtil.CURR_TIME_STR}] AD is not valid.\n")
                        return@OnClickListener
                    }
                    rewardedAdManager?.showAd(requireActivity())
                }
            )
        }

        val adParam = AdParam.Builder()
            .setAdUnitId(AD_UNIT_ID)
            .build()
        rewardedAdManager = GfpRewardedAdManager(requireActivity(), adParam)

        rewardedAdManager?.setAdListener(
            object : RewardedAdListener() {
                override fun onAdLoaded(ad: GfpRewardedAd) {
                    logTextView.append(
                        "[${DateUtil.CURR_TIME_STR}] AD Loaded.\n\t\tAdProviderName: ${ad.adProviderName}\n" // ktlint-disable max-line-length
                    )
                    showButton.isEnabled = true
                }

                override fun onAdStarted(ad: GfpRewardedAd) {
                    logTextView.append("[${DateUtil.CURR_TIME_STR}] AD Started.\n")
                }

                override fun onAdClicked(ad: GfpRewardedAd) {
                    logTextView.append("[${DateUtil.CURR_TIME_STR}] AD Clicked.\n")
                }

                override fun onAdClosed(ad: GfpRewardedAd) {
                    logTextView.append("[${DateUtil.CURR_TIME_STR}] AD Closed.\n")
                    showButton.isEnabled = false
                }

                override fun onAdCompleted(ad: GfpRewardedAd) {
                    logTextView.append("[${DateUtil.CURR_TIME_STR}] AD Completed.\n")
                }

                override fun onError(ad: GfpRewardedAd, error: GfpError) {
                    logTextView.append(
                        "[${DateUtil.CURR_TIME_STR}] Error occurred.\n\tcode[${error.errorCode}]\n\tsubCode[${error.errorSubCode}]\n\t\tmessage[${error.errorMessage}]\n" // ktlint-disable max-line-length
                    )
                }
            }
        )
        logTextView.append("[${DateUtil.CURR_TIME_STR}] AD Requested.\n")
        rewardedAdManager?.loadAd()
        return binding.root
    }

    override fun onDestroyView() {
        super.onDestroyView()
        rewardedAdManager?.destroy()
    }

    companion object {
        private const val AD_UNIT_ID = "aos_rv_test"
    }
}
